src/util/gas-factory-helpers.ts

```diff
@@ -543,3 +544,161 @@ export function initSwapRouteFromExisting(
     portionAmount: swapRoute.portionAmount,
   };
 }
+
+export const calculateL1GasFeesHelper = async (
+  route: RouteWithValidQuote[],
+  chainId: ChainId,
+  usdPool: Pair | Pool,
+  quoteToken: Token,
+  nativePool: Pair | Pool | null,
+  l2GasData?: ArbitrumGasData | OptimismGasData
+): Promise<{
+  gasUsedL1: BigNumber;
+  gasUsedL1OnL2: BigNumber;
+  gasCostL1USD: CurrencyAmount;
+  gasCostL1QuoteToken: CurrencyAmount;
+}> => {
+  const swapOptions: SwapOptionsUniversalRouter = {
+    type: SwapType.UNIVERSAL_ROUTER,
+    recipient: '0x0000000000000000000000000000000000000001',
+    deadlineOrPreviousBlockhash: 100,
+    slippageTolerance: new Percent(5, 10_000),
+  };
+  let l1Used = BigNumber.from(0);
+  let l1FeeInWei = BigNumber.from(0);
+  let gasUsedL1OnL2 = BigNumber.from(0);
+  const opStackChains = [
+    ChainId.OPTIMISM,
+    ChainId.OPTIMISM_GOERLI,
+    ChainId.OPTIMISM_SEPOLIA,
+    ChainId.BASE,
+    ChainId.BASE_GOERLI,
+  ];
```

Comment:
would prefer this in constants file and naming to be something like `L2_FEE_CHAINS` or something more generic